File: src/AutoRest.SdkExplorer/Model/OpenAi/AiChatSession.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;

namespace AutoRest.SdkExplorer.Model.OpenAi
{
    public class AiChatSession
    {
        public string SessionId { get; set; } = string.Empty;
        public List<AiChatMessage> ChatMessages { get; set; } = new List<AiChatMessage>();
        public List<AiFunctionDefinition> Functions { get; set; } = new List<AiFunctionDefinition>();

        public float? NucleusSamplingFactor { get; set; }
        public float? PresencePenalty { get; set; }
        public float? FrequencyPenalty { get; set; }
        public float? Temperature { get; set; }

        public AiChatSession()
        {
        }

        public AiChatSession(string sessionId)
        {
            this.SessionId = sessionId;
        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Ai/AiChatSession.ts
import { AiChatMessage } from "./AiChatMessage";
import { AiFunctionDefinition } from "./AiFunctionDefinition";

export class AiChatSession implements AutoRest.SdkExplorer.Interface.AiChatSession {
    sessionId?: string;
    chatMessages?: AiChatMessage[];
    functions?: AiFunctionDefinition[];

    nucleusSamplingFactor? : number;
    presencePenalty? : number;
    frequencyPenalty?: number;
    temperature?: number;

    constructor(data: AutoRest.SdkExplorer.Interface.AiChatSession) {
        this.sessionId = data.sessionId;
        this.chatMessages = data.chatMessages?.map(m => new AiChatMessage(m));
        this.functions = data.functions?.map(f => new AiFunctionDefinition({
            name: f.name,
            description: f.description,
            parametersAsString: f.parametersAsString
        }));
    }
}
